torch/testing/_internal/common_device_type.py

```diff
@@ -205,8 +213,63 @@ def _get_precision_override(self, test, dtype):
     # Creates device-specific tests.
     @classmethod
     def instantiate_test(cls, name, test):
-        test_name = name + "_" + cls.device_type
+        # Handles tests using the ops decorator
+        if hasattr(test, "op_list"):
+            for op in test.op_list:
+                base_op_test_name = name + "_" + op.name + "_" + cls.device_type
```

Comment:
As the algorithm for computing the name of the test gets more and more complicated, I get more worried for developers who  write a test, and then want to run it, and find that `TestFoo.test_blah` which is the class and method they wrote doesn't work anymore. I'm wondering if we can help users out: suppose someone requests a test with the generic name (and not the op specific name). It would be really helpful if the test suite reported "Hey, this is a generic test, here are all of the specific instantiations you could ask for". Or maybe it should go ahead and run each variant? Not sure.

Reply:
This doesn't have to be in this PR but I think it's definitely worth following up on.

Reply:
I agree. I actually had that in the early design for the device generic framework but we thought that filtering tests by using pytest was OK. The Wiki suggests using a command like `pytest test_torch.py -k 'test_diagonal'`, for example, to run all the variants of a test named `test_diagonal`.

Reply:
Not everyone uses pytest. It's also better to avoid pytest if you're debugging a segfault. And some of our test files literally do not support pytest. Also, filtering for `test_diagonal` doesn't even work, because it's gotta be `test_testname_diagonal`, right? I think looking for opportunities to make the UI self documenting will pay off.

Reply:
Oh, sorry, I was unclear. The test the Wiki talks about is actually called `test_diagonal`. Yes, with op instantiation you would have to filter like you're describing. 

I made a note to follow-up on your proposal.

Reply:
`-k` works with `python` now too, not just `pytest`.